core/utilities/format_stint_time.py

```
@@ -0,0 +1,47 @@
+        base = stint_time.split('.', 1)[0]
+        # valid time patterns are H:MM or H:MM:SS (hours can be multiple digits)
+        if re.match(r"^\d+:[0-5]\d(?::[0-5]\d)?$", base):
+            return base
+        # otherwise try treating the string as a numeric seconds value
+        try:
+            total_seconds = max(int(float(stint_time)), 0)
+            hours = total_seconds // 3600
+            minutes = (total_seconds % 3600) // 60
+            seconds = total_seconds % 60
+            return f"{hours:02d}:{minutes:02d}:{seconds:02d}"
+        except Exception:
```

Ruff (0.15.4)

[warning] Do not catch blind exception: `Exception`

(BLE001)